### PR TITLE
Redfish refactor

### PR DESCRIPTION
**Pull request by dalebremner**

### CODE REVIEW

File: spec/lib/api/redfish-1.0/dcimcooling-spec.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                    '/redfish/v1/DCIMCooling/' + domain + '/' + type +
+                    '/' + coolingNode.identifiers[0] + '-' + coolingNode.id);
+            });
+    });
+
+    it('/DCIMCooling/{domain}/{type}/{identifier} should return a valid cooling id', function () {
+        waterline.nodes.needByIdentifier.resolves([coolingNode]);
+        var domain = coolingNode.identifiers[2];
+        var type = coolingNode.identifiers[3];
+        var identifier = coolingNode.id;
```
[the remark this answers is not in the document]
Comment (dalebremner) [reply spec/lib/api/redfish-1.0/dcimcooling-spec.js]: done

File: spec/lib/api/redfish-1.0/dcimpower-spec.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                    '/redfish/v1/DCIMPower/' + domain + '/' + type +
+                    '/' + powerNode.identifiers[0] + '-' + powerNode.id);
+            });
+    });
+
+    it('/DCIMPower/{domain}/{type}/{identifier} should return a valid power id', function () {
+        waterline.nodes.needByIdentifier.resolves([powerNode]);
+        var domain = powerNode.identifiers[2];
+        var type = powerNode.identifiers[3];
+        var identifier = powerNode.id;
```
[the remark this answers is not in the document]
Comment (srinia6) [reply spec/lib/api/redfish-1.0/dcimpower-spec.js]: Done

File: spec/lib/api/redfish-1.0/networks-spec.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+describe('Redfish Networks', function () {
+    var redfish;
+    var waterline;
+    var Promise;
+    var view;
+    var fs;
+    var wsman;
+    var systems;
+
+    var dellNodeObm ={
```
[the remark this answers is not in the document]
Comment (srinia6) [reply spec/lib/api/redfish-1.0/networks-spec.js]: done

File: spec/lib/api/redfish-1.0/networks-spec.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    it('/NetworkDevices/{identifier} should return a valid network device id for Redfish endpoints', function () {
+        waterline.nodes.find.resolves([NetworksNode]);
+        this.sandbox.spy(redfish, 'render');
+        redfish.getRedfishNetworkCatalog.resolves(networkSwitchCatalog);
+        wsman.isDellSystem.resolves({node: NetworksNode, isDell: false, isRedfishCapable: false});
+        return helper.request().get('/redfish/v1/NetworkDevices/' +
+            NetworksNode.identifiers[0] + '-' + NetworksNode.id)
+            .expect('Content-Type', /^application\/json/)
+            .expect(200)
+            .expect(function(res) {
```
[the remark this answers is not in the document]
Comment (srinia6) [reply spec/lib/api/redfish-1.0/networks-spec.js]: done

File: spec/lib/api/redfish-1.0/networks-spec.js [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+    it('/NetworkDevices/{identifier} should return a valid network device id for Dell endpoints', function () {
+        this.sandbox.stub(redfish, 'render');
+        waterline.nodes.find.resolves([NetworksNode]);
+        redfish.getRedfishNetworkCatalog.resolves(networkSwitchCatalog);
+        wsman.isDellSystem.resolves({node: dellNode, isDell: true, isRedfishCapable: false});
+        return helper.request().get('/redfish/v1/NetworkDevices/' +
+            NetworksNode.identifiers[0] + '-' + NetworksNode.id)
+            .expect(200)
+            .expect(function(res) {
```
[the remark this answers is not in the document]
Comment (srinia6) [reply spec/lib/api/redfish-1.0/networks-spec.js]: done

File: spec/lib/api/redfish-1.0/networks-spec.js [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+    });
+
+    it('/NetworkDevices/{identifier} should return a valid network device id for Quanta endpoints', function () {
+        waterline.nodes.find.resolves([NetworksNode]);
+        this.sandbox.stub(redfish, 'render');
+        redfish.getRedfishNetworkCatalog.resolves(networkSwitchCatalog);
+        wsman.isDellSystem.resolves({node: NetworksNode, isDell: false, isRedfishCapable: false});
+        return helper.request().get('/redfish/v1/NetworkDevices/' + NetworksNode.id)
+            .expect(200)
+            .expect(function(res) {
```
[the remark this answers is not in the document]
Comment (srinia6) [reply spec/lib/api/redfish-1.0/networks-spec.js]: done

File: lib/services/redfish-api-service.js [hunk 7]
```diff
@@ -209,7 +211,30 @@ function redfishApiServiceFactory(
...
+    RedfishApiService.prototype.getRedfishCatalog = function (req, res) {
+        var identifier = req.swagger.params.identifier.value;
+        var nodeId = identifier.split(['-'])[1];
+        var typeId = identifier.split(['-'])[0];
+        var newId = typeId.replace(req.swagger.params.type.raw, '');
+        var catalogSource = req.url.replace(req.swagger.params.identifier.value, newId);
+
+        return nodeApi.getNodeCatalogSourceById(nodeId, catalogSource)
+            .catch(function (error) {
+                return redfish.handleError(error, res);
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/services/redfish-api-service.js]: done

File: lib/services/redfish-api-service.js [hunk 8]
```diff
@@ -209,7 +211,30 @@ function redfishApiServiceFactory(
...
+    };
+    RedfishApiService.prototype.getRedfishNetworkCatalog = function (req, res) {
+        var identifier = req.swagger.params.identifier.value;
+        var nodeId = identifier.split(['-'])[1];
+        var networkId = identifier.split(['-'])[0];
+        var catalogSource = req.url.replace(req.swagger.params.identifier.value, networkId);
 
+        return nodeApi.getNodeCatalogSourceById(nodeId, catalogSource)
+            .catch(function (error) {
+                return redfish.handleError(error, res);
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/services/redfish-api-service.js]: done

File: lib/api/redfish-1.0/networks.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+var redfish = injector.get('Http.Api.Services.Redfish');
+var waterline = injector.get('Services.Waterline');
+var Promise = injector.get('Promise');
+var _ = injector.get('_');
+var controller = injector.get('Http.Services.Swagger').controller;
+var Errors = injector.get('Errors');
+var moment = require('moment');
+var wsman = injector.get('Http.Services.Wsman');
+var systems = require('./systems.js');
+var mktemp = require('mktemp');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/networks.js]: done

File: lib/api/redfish-1.0/networks.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var redfish = injector.get('Http.Api.Services.Redfish');
+var waterline = injector.get('Services.Waterline');
+var Promise = injector.get('Promise');
+var _ = injector.get('_');
+var controller = injector.get('Http.Services.Swagger').controller;
+var Errors = injector.get('Errors');
+var moment = require('moment');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/networks.js]: Done

File: lib/api/redfish-1.0/networks.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var redfish = injector.get('Http.Api.Services.Redfish');
+var waterline = injector.get('Services.Waterline');
+var Promise = injector.get('Promise');
+var _ = injector.get('_');
+var controller = injector.get('Http.Services.Swagger').controller;
+var Errors = injector.get('Errors');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/networks.js]: done

File: lib/api/redfish-1.0/networks.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+var moment = require('moment');
+var wsman = injector.get('Http.Services.Wsman');
+var systems = require('./systems.js');
+var mktemp = require('mktemp');
+var fs = require('fs');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/networks.js]: done

File: lib/api/redfish-1.0/dcimpower.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
+// Copyright © 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var redfish = injector.get('Http.Api.Services.Redfish');
+var Promise = injector.get('Promise'); // jshint ignore:line
+var _ = injector.get('_');  // jshint ignore:line
+var controller = injector.get('Http.Services.Swagger').controller;
+var Errors = injector.get('Errors');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/dcimpower.js]: done

File: lib/api/redfish-1.0/dcimpower.js [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+var workflow = injector.get('Http.Services.Api.Workflows');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/dcimpower.js]: done

File: lib/api/redfish-1.0/dcimpower.js [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+var waterline = injector.get('Services.Waterline');
+var nodeApi = injector.get('Http.Services.Api.Nodes');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/dcimpower.js]: done

File: lib/api/redfish-1.0/dcimpower.js [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+var Logger = injector.get('Logger');
+var logger = Logger.initialize('event-service');
```
[the remark this answers is not in the document]
Comment (srinia6) [reply lib/api/redfish-1.0/dcimpower.js]: done

**Comment by srinia6** [comment]
test this please

**Pull request merged**